unit-tests/shared/src/test/scala/org/scalanative/testsuite/javalib/util/concurrent/LinkedTransferQueueTest.scala

```diff
@@ -0,0 +1,1034 @@
+  }
+
+  /** toArray(a) contains all elements in FIFO order //
+   */
+  @Test def testToArray2() = {
+    val q = populatedQueue(SIZE)
+    val items = new Array[Item](SIZE)
+    val array = q.toArray(items)
+    assertSame(items, array)
+    for (o <- items)
+      assertSame(o, q.poll())
+    assertTrue(q.isEmpty())
+  }
+
+  /** toArray(incompatible array type) throws ArrayStoreException
+   *
+   *  We don't have this, because SN doesn't yet do runtime variance check.
+   */
+  // @Test def testToArray_incompatibleArrayType() = {
+  //   val q = populatedQueue(SIZE)
+  //   try {
+  //     val ss: Array[String] = q.toArray(new Array[String](10))
+  //     shouldThrow()
+  //   } catch {
+  //     case _: ArrayStoreException => {}
+  //   }
+  // }
+
+  /** iterator iterates through all elements
+   */
+  @Test def testIterator() = {
+    val q = populatedQueue(SIZE)
+    var it = q.iterator()
+    var i = 0
+    while (it.hasNext()) {
+      mustContain(q, it.next())
+      i += 1
+    }
+    mustEqual(i, SIZE)
+    assertIteratorExhausted(it)
+
+    it = q.iterator()
+    i = 0
+    while (it.hasNext()) {
+      mustEqual(it.next(), q.take())
+      i += 1
+    }
+    mustEqual(i, SIZE)
+    assertIteratorExhausted(it)
+  }
+
+  /** iterator of empty collection has no elements
+   */
+  @Test def testEmptyIterator() = {
+    assertIteratorExhausted(new LinkedTransferQueue[Item]().iterator())
+  }
+
+  /** iterator.remove() removes current element
+   */
+  @Test def testIteratorRemove() = {
+    val q = new LinkedTransferQueue[Item]()
+    q.add(itemFor(two))
+    q.add(itemFor(one))
+    q.add(itemFor(three))
+
+    var it = q.iterator()
+    it.next()
+    it.remove()
+
+    it = q.iterator()
+    assertSame(it.next(), itemFor(one))
+    assertSame(it.next(), itemFor(three))
+    assertFalse(it.hasNext())
+  }
+
+  /** iterator ordering is FIFO
+   */
+  @Test def testIteratorOrdering() = {
+    val q = new LinkedTransferQueue[Item]()
+    mustEqual(Integer.MAX_VALUE, q.remainingCapacity())
+    q.add(itemFor(one))
+    q.add(itemFor(two))
+    q.add(itemFor(three))
+    mustEqual(Integer.MAX_VALUE, q.remainingCapacity())
+    var k = 0
+    val it = q.iterator()
+    while (it.hasNext()) {
+      k += 1
+      mustEqual(k, it.next())
+    }
+    mustEqual(3, k)
+  }
+
+  /** Modifications do not cause iterators to fail
+   */
+  @Test def testWeaklyConsistentIteration() = {
+    val q = new LinkedTransferQueue[Item]()
+    q.add(itemFor(one))
+    q.add(itemFor(two))
+    q.add(itemFor(three))
+    val it = q.iterator()
+    while (it.hasNext()) {
+      q.remove()
+      it.next()
+    }
+    mustEqual(0, q.size())
+  }
+
+  /** offer transfers elements across Executor tasks
+   */
+  @Test def testOfferInExecutor(): Unit = {
+    val q = new LinkedTransferQueue[Item]()
+    val threadsStarted = new CheckedBarrier(2)
+    val executor = Executors.newFixedThreadPool(2)
+    Using(cleaner(executor)) { cleaner =>
```

Comment:
For the sake of Scala 2.12 compability there is defined a helper method for these in JSR166Test: ` usingPoolCleaner(new ForkJoinPool()) { p: ForkJoinPool =>`